File: nodejs/pay.js
var https = require('https');
var getCurrencies = require('./currencies');

var usdAmount = 10;
var currencies = ["ETH"];

getReq(usdAmount, currencies, function(data) {
  console.log(JSON.stringify(data.request)); //return this data to PayBear form
});

function getReq(amount, currencies, callback) {
  var orderId = 1;

  var request = {
    currencies: [],
    invoice: ""+orderId,
    fiatValue: +amount,
    accepted: false,
    redirectTo: 'https://www.mysite.com/success.html'
  };

  sendReq(orderId, function(data) {
    if(data.address) {
      getCurrencies(currencies, data.address, request, function(result) {
        request.currencies = result;
        callback({
          error: null,
          request: request
        });
      });
    } else {
      callback({
        error: data.error,
        request: null
      });
    }
  });
}

function sendReq(orderId, callback) {
  var payoutAddress = '0x39ee76948d238fad2b750998f8a38d80c73c7cd7'; //your address
  var callbackUrl = 'https://www.mysite.com/payBear/callback/' + orderId;
  var url = 'https://api.paybear.io/v1/eth/payment/' + payoutAddress + '/' + encodeURIComponent(callbackUrl);

  https.get(url, function (res) {
    var rawData = '';
    res.on('data', function (chunk) { rawData += chunk; });

    res.on('end', function () {
      var response = JSON.parse(rawData);
      if (response.success) {
        callback({
          error: null,
          address: response.data.address
        });
      }
    });
  }).
  on('error', function (e) {
    console.error(e);
    callback({
      error: e,
      address: null
    });
  });
}

File: rails/pay.rb
require 'currencies.rb'

def pay
  usd_amount = 10
  currencies = ['ETH']
  puts get_req(usd_amount, currencies)[:request] # return this data to PayBear form
end

def get_req(amount, currencies)
  order_id = 1

  request = {
    currencies: [],
    invoice: order_id,
    fiatValue: amount,
    accepted: false,
    redirectTo: 'https://www.mysite.com/success.html'
  }

  data = send_req(order_id)

  if data[:address]
    result = get_currencies(currencies, data[:address], request)
    request[:currencies] = result

    {
      error: nil,
      request: request
    }
  else
    {
      error: data[:error],
      request: nil
    }
  end
end

def send_req(order_id)
  payout_address = '0x39ee76948d238fad2b750998f8a38d80c73c7cd7' # your address
  callback_url = "http://CHANGEME.com/callback/#{order_id}"
  url = "https://api.paybear.io/v1/eth/payment/#{payout_address}/#{CGI.escape(callback_url)}"

  begin
    response = ActiveSupport::JSON.decode(open(url).read)
    rescue *HTTP_ERRORS => e
    {
      error: e,
      address: nil
    }
  else
    if response['success']
      {
        error: nil,
        address: response['data']['address']
      }
    end
  end
end